tuna/populate_golden.py

```diff
@@ -0,0 +1,155 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""! @brief Script to populate the golden table based on session_id"""
+from tuna.parse_args import TunaArgs, setup_arg_parser
+from tuna.utils.logger import setup_logger
+from tuna.tables import DBTables
+from tuna.dbBase.sql_alchemy import DbSession
+from sqlalchemy.exc import IntegrityError, OperationalError  #pylint: disable=wrong-import-order
+
+# Setup logging
+LOGGER = setup_logger('populate_golden')
+
+
+def parse_args():
+  """! Function to parse arguments"""
+  parser = setup_arg_parser('Populate golden table based on session_id',
+                            [TunaArgs.CONFIG_TYPE])
+  parser.add_argument(
+      '--session_id',
+      dest='session_id',
+      action='store',
+      type=int,
+      required=True,
+      help=
+      'Session ID to be used as tuning tracker. Allows to correlate DB results to tuning sessions'
+  )
+  parser.add_argument('--golden_v',
+                      dest='golden_v',
+                      action='store',
+                      type=int,
+                      default=None,
+                      required=False,
+                      help='Golden miopen version')
+  args = parser.parse_args()
+  return args
+
+
+def get_fdb_query(dbt):
+  """! Compose query to get all fdb entries for the session"""
+  with DbSession() as session:
+    query = session.query(dbt.find_db_table)\
+            .filter(dbt.find_db_table.session == dbt.session.id)\
+            .filter(dbt.find_db_table.kernel_time != -1)\
+            .filter(dbt.find_db_table.valid == 1)
+
+  return query
+
+
+def get_golden_query(dbt, golden_version):
+  """! Compose query to get all entries for a golden miopen version"""
+  with DbSession() as session:
+    query = session.query(dbt.golden_table)\
+            .filter(dbt.golden_table.golden_miopen_v == golden_version)\
+            .filter(dbt.find_db_table.valid == 1)
+
+  return query
+
+
+def latest_golden_v(dbt):
+  """Get highest golden version in the table """
+  version = -1
+  with DbSession() as session:
+    query = session.query(dbt.golden_table).order_by(
+        dbt.golden_table.golden_miopen_v.desc()).limit(1)
```

Comment:
in worst case scenario using max is faster than sorting, please use the max function to extract the highest id.